nibabel/funcs.py

```diff
@@ -88,43 +88,81 @@ def squeeze_image(img):
                  img.extra)
 
 
-def concat_images(images, check_affines=True):
-    ''' Concatenate images in list to single image, along last dimension
+def _shape_equal_excluding(shape1, shape2, exclude_axes):
+    """ Helper function to compare two array shapes, excluding any
+    axis specified."""
+
+    if len(shape1) != len(shape2):
+        return False
+
+    idx_mask = np.ones((len(shape1),), dtype=bool)
+    idx_mask[exclude_axes] = False
+    return np.array_equal(np.asarray(shape1)[idx_mask],
+                          np.asarray(shape2)[idx_mask])
+
+
+def concat_images(images, check_affines=True, axis=None):
+    ''' Concatenate images in list to single image, along specified dimension
 
     Parameters
     ----------
     images : sequence
-       sequence of ``SpatialImage`` or of filenames\s
+       sequence of ``SpatialImage`` or filenames of the same dimensionality\s
     check_affines : {True, False}, optional
        If True, then check that all the affines for `images` are nearly
        the same, raising a ``ValueError`` otherwise.  Default is True
-
+    axis : None or int, optional
+        If None, concatenates on a new dimension.  This rrequires all images
+           to be the same shape).
+        If not None, concatenates on the specified dimension.  This requires
+           all images to be the same shape, except on the specified dimension.
+           For 4D images, axis must be between -2 and 3.
```

Comment:
Is this last sentence necessary - I mean - won't the concatenate raise an informative error in cases outside this range?